src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -531,6 +531,16 @@ public DatasetSummary retrieveSummaryByName(String name) {
         if (!whereClauses.isEmpty()) {
             whereSql = " WHERE " + StringUtils.join(whereClauses, " AND ");
         }
+
+        // get the filtered total count of objects without offset and limit
+        String filteredTotalSql = "SELECT count(id) AS total FROM dataset WHERE " +
```

Comment:
We should just keep an eye on scale here.  It's fine for now but this does require multiple round trips to the DB

Reply:
Yeah, it could be condensed into something with multiple subqueries, but these aren't particularly complex queries.

Reply:
I was going to make the same point but I realized we're already doing two DB calls in this method, so what's one more? Although I'm pretty sure there's a SQL way to get both all three results in a single query but I'd have to google to find it.

Reply:
There is with windowing functions but it gets pretty opaque pretty fast :)

Reply:
I'd say we optimize for it when it becomes a problem...not thinking we'll hit 100K datasets / snapshot for a while